python/ray/tests/test_chaos.py

```diff
@@ -125,6 +126,36 @@ def get(self):
     # assert_no_system_failure(p, 10000, 10)
 
 
+@pytest.mark.skip(reason="Flaky.")
+@pytest.mark.parametrize(
+    "ray_start_chaos_cluster", [{
+        "kill_interval": 8,
+        "env_vars": {
+            "RAY_lineage_pinning_enabled": "true"
+        },
+        "head_resources": {
+            "CPU": 1
+        },
+        "worker_node_types": {
+            "cpu_node": {
+                "resources": {
+                    "CPU": 8,
+                },
+                "node_config": {},
+                "min_workers": 0,
+                "max_workers": 4,
+            },
+        },
+    }],
+    indirect=True)
+def test_chaos_lineage_reconstruction(ray_start_chaos_cluster):
```

Comment:
This seems to be

1. sometimes hanging when reconstruction is on.
2. consistently hanging (instead of raising an exception) when reconstruction is off.

The expected behavior is that 1. it should always pass, 2. it should raise an exception.

cc @stephanie-wang it also seems like the observability during reconstruction is not that great. It was really hard to figure out what was going on when it was hanging. Maybe we should think about how to improve this part as well.

Reply:
I will add dask on ray nightly test once this tests consistently pass.

Reply:
Hmm it sounds like a bug. Can you open an issue? You can leave this test out of the PR for now.